src/Builder/BuilderEncoder.php

```diff
@@ -129,6 +130,10 @@ public function encode($value): stdClass|array|string
      */
     private function encodeAsArray(OperatorInterface $value): stdClass
     {
+        if ($value instanceof BackedEnum) {
+            return $this->wrap($value, [$value->value]);
```

Comment:
Is the NoInterfaceProperties error below the result of Psalm failing to infer that BackedEnum has an implicit `$value` property?

Reply:
Yes it look like an issue in palm: https://github.com/vimeo/psalm/issues/10585